lib/zendesk_apps_support/validations/requests.rb

```diff
@@ -1,57 +1,69 @@
 # frozen_string_literal: true
 
-require 'uri'
 require 'ipaddress_2'
 
 module ZendeskAppsSupport
   module Validations
     module Requests
       class << self
-        HTTP_REQUEST_METHODS = %w[GET HEAD POST PUT DELETE CONNECT OPTIONS TRACE].freeze
-
-        REQUEST_CALLS = [
-          /\w+\.request\(['"](.*)['"]/i, # ZAF request
-          /(?:\$|jQuery)\.(?:ajax|get|post|getJSON)\(['"](.*)['"]/i, # jQuery request
-          /\w+\.open\(['"](?:#{Regexp.union(HTTP_REQUEST_METHODS).source})['"],\s?['"](.*)['"]/i, # XMLHttpRequest
-          /fetch\(['"](.*)['"]/i # fetch
-        ].freeze
+        HTTP_URI = /[\'\"](http[^s].*?)[\'\"]/i
```

Comment:
Are the `[\'\"]` blocks needed here? 

Also, might be better to just use the [URI::regexp](https://apidock.com/ruby/URI/regexp/class) for this...

Reply:
+1 on using URI::regexp

Reply:
Yeah this will work

```
(byebug) file_content.scan(URI.regexp(["http"]))
[["http", "hello.com'", nil, nil, nil, nil, nil, nil, nil], ["http", "wikipedia.org'", nil, nil, nil, nil, nil, nil, nil], ["http", nil, nil, "www.foo-bar.com", nil, nil, nil, nil, nil]]

(byebug) file_content.scan(URI.regexp(["http"])).map(&:compact).map(&:last)
["hello.com'", "wikipedia.org'", "www.foo-bar.com"]
```

`.match` only returns the first one it finds
```
(byebug) file_content.match(URI.regexp(['http']))
#<MatchData "http:hello.com'" 1:"http" 2:"hello.com'" 3:nil 4:nil 5:nil 6:nil 7:nil 8:nil 9:nil>
```

Reply:
Thank you for the suggestion on using `URI.regexp(['http'])`, @JeanMarcGoepfert .